src/java/org/apache/nutch/crawl/CrawlDb.java

```diff
@@ -111,7 +116,7 @@ public void update(Path crawlDb, Path[] segments, boolean normalize,
       LOG.info("CrawlDb update: Merging segment data into db.");
     }
     try {
-      JobClient.runJob(job);
+      int complete = job.waitForCompletion(true)?0:1;
     } catch (IOException e) {
```

Comment:
Need to catch all thrown exceptions, incl. ClassNotFoundException and InterruptedException. Could also just catch `Exception`.